addons/im_livechat/controllers/main.py

```diff
@@ -12,18 +13,40 @@ class LivechatController(http.Controller):
     # Note: the `cors` attribute on many routes is meant to allow the livechat
     # to be embedded in an external website.
 
-    @http.route('/im_livechat/external_lib.<any(css,js):ext>', type='http', auth='public')
-    def livechat_lib(self, ext, **kwargs):
-        # _get_asset return the bundle html code (script and link list) but we want to use the attachment content
-        bundle = 'im_livechat.external_lib'
+    @http.route('/im_livechat/external_lib.<any(css,js):ext>', type='http', auth='public', cors='*')
+    def external_lib(self, ext, **kwargs):
+        """ Preserve compatibility with legacy livechat imports. Only
+        serves javascript since the css will be fetched by the shadow
+        DOM of the livechat to avoid conflicts.
+        """
+        bundle = 'im_livechat.assets_embed'
+        files, _ = request.env["ir.qweb"]._get_asset_content(bundle)
+        asset = AssetsBundle(bundle, files)
+        if ext == 'css':
+            raise request.not_found()
```

Comment:
this could had been simplified by changing the route to `/im_livechat/external_lib.js`